### PR TITLE
Handle pod failures / Add k8s clientset to Reporter struct

### PR DESCRIPTION
**Pull request by SoroushTaheri**

### CODE REVIEW

File: pkg/reporter/reporter.go [hunk 1]
```diff
@@ -67,6 +67,20 @@ func (r *Reporter) Report(ctx context.Context, room *hubv1.Room) (bool, error) {
...
+		Id:     room.Spec.ID,
+		Status: api.Result_failed,
+		Msg:    fmt.Sprintf("Timeout limit exceeded (%d seconds).", threshold),
+	}
+	// TODO: should write better result for backend
+	if err := r.informRabbit(room, result); err != nil {
+		return err
+	}
+	return nil
+
```
Comment (AliMirlou) [inline pkg/reporter/reporter.go:81]: This new line is redundant.

File: pkg/timer/timer.go [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+			t.log.WithValues("podName", actorPodName).Info("Timer for pod exists.")
+		} else {
+			t.timers[actorPodName] = room.Spec.Timeout
+			go t.startPodTimer(actorPodName, room)
+		}
+	}
+}
+
+// startPodTimer measures the age of a running actor/director pod and kills the room if a pod's age exceeds the given limit.
+// Please note that this function DOES return an error object (if there's any) but because it's supposed to run as a goroutine you cannot actually receive/observe the given error. Error handling of this method needs some work.
```
Comment (AliMirlou) [inline pkg/timer/timer.go:68]: Put a TODO on this line.

**Review by AliMirlou — CHANGES_REQUESTED** [review]
Left some small comments to consider.

File: pkg/reporter/reporter.go (hunk 1 above)
Comment (SoroushTaheri) [reply pkg/reporter/reporter.go]: Fixed in bd976e5

File: pkg/timer/timer.go (hunk 2 above)
Comment (SoroushTaheri) [reply pkg/timer/timer.go]: Fixed in 7243615

**Review by AliMirlou — APPROVED** [review]
LGTM

**Pull request merged**